### PR TITLE
Upgrade to Castle.Windsor 5.0.0

### PR DESCRIPTION
**Pull request by danielmarbach**
Replaces #60

### CODE REVIEW

File: src/NServiceBus.CastleWindsor.AcceptanceTests/NServiceBus.CastleWindsor.AcceptanceTests.csproj [hunk 1]
```diff
@@ -10,7 +10,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Castle.Windsor" Version="4.*" />
+    <PackageReference Include="Castle.Windsor" Version="5.*" />
```
Comment (bording) [inline src/NServiceBus.CastleWindsor.AcceptanceTests/NServiceBus.CastleWindsor.AcceptanceTests.csproj]: We should not be using wildcards in package references.

File: src/NServiceBus.CastleWindsor.Tests/NServiceBus.CastleWindsor.Tests.csproj [hunk 2]
```diff
@@ -12,7 +12,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Castle.Windsor" Version="4.*" />
+    <PackageReference Include="Castle.Windsor" Version="5.*" />
```
Comment (bording) [inline src/NServiceBus.CastleWindsor.Tests/NServiceBus.CastleWindsor.Tests.csproj]: We should not be using wildcards in package references.

**Comment by tmasternak** [comment]
@bording please have a look now.

**Comment by timbussmann** [comment]
👍 for referencing 5.0.1 in the tests, not sure we need to target 5.0.1 in the main package though as I'm not sure there is a critical fix in there?

**Pull request merged**